File: java-e2e/src/test/java/com/testflow/e2e/pages/BasePage.java
package com.testflow.e2e.pages;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.LoadState;

public abstract class BasePage {

    protected static final double TIMEOUT_SHORT = 5_000;
    protected static final double TIMEOUT_MEDIUM = 10_000;

    protected final Page page;

    protected BasePage(Page page) {
        this.page = page;
    }

    protected void navigate(String path) {
        page.navigate(path);
    }

    protected void waitForNetworkIdle() {
        page.waitForLoadState(LoadState.NETWORKIDLE);
    }
}

File: e2e-bdd/pages/base.page.ts
import { Page } from '@playwright/test';

export class BasePage {
  protected readonly TIMEOUT_SHORT  = 5_000;
  protected readonly TIMEOUT_MEDIUM = 10_000;

  constructor(protected page: Page) {}

  async navigate(path: string): Promise<void> {
    await this.page.goto(path);
  }

  async waitForNetworkIdle(): Promise<void> {
    await this.page.waitForLoadState('networkidle');
  }
}
